Sources/SwifterSwift/SwiftStdlib/StringExtensions.swift

```diff
@@ -606,23 +606,45 @@ public extension String {
     ///		"Hello World!"[safe: 6..<11] -> "World"
     ///		"Hello World!"[safe: 21..<110] -> nil
     ///
-    /// - Parameter range: Half-open range.
-    subscript(safe range: CountableRange<Int>) -> String? {
-        guard let lowerIndex = index(startIndex, offsetBy: max(0, range.lowerBound), limitedBy: endIndex) else { return nil }
-        guard let upperIndex = index(lowerIndex, offsetBy: range.upperBound - range.lowerBound, limitedBy: endIndex) else { return nil }
-        return String(self[lowerIndex..<upperIndex])
+    /// - Parameter substringRange: Half-open range.
+    subscript(safe substringRange: CountableRange<Int>) -> String? {
+        guard
+            !isEmpty,
+            substringRange.lowerBound >= 0,
+            let lowerIndex = index(startIndex, offsetBy: substringRange.lowerBound, limitedBy: endIndex),
+            let upperIndex = index(lowerIndex, offsetBy: substringRange.upperBound - substringRange.lowerBound, limitedBy: endIndex),
+            let range = range(of: self) else {
+                return nil
+        }
+
+        if range ~= lowerIndex {
```

Comment:
Why do you need this? You're already checking `!isEmpty` and `substringRange.lowerBound >= 0`.

Reply:
Without `if range ~= lowerIndex` this test case fails `XCTAssertNil(oneCharStr[safe: 1..<1])`. It is returning empty string instead of `nil`

Reply:
Strangely this is the expected outcome in Swift. If you check what happens in `Array`, for example, you'll see that you can access the `endIndex` to get back an empty `ArraySlice`
```Swift
let oneNumArr = [0]
oneNumArr[0..<0] // ArraySlice([])
oneNumArr[1..<1] // ArraySlice([])
oneNumArr[2..<2] // Fatal error: Array index is out of range
```
I think for the sake of consistency, even though I agree that it doesn't make sense, we should maintain this result.

Reply:
I've also added a few test case so that it matches:
```
let emptyArr: [Int] = []
emptyArr[0..<0] // ArraySlice([])
emptyArr[0..<1] // Fatal error: Array index is out of range
```